### PR TITLE
Replaces rust bcrypt tool with a java implementation

### PR DESCRIPTION
**Pull request by soenkeliebau**
this is necessary for now due to https://github.com/stackabletech/stackable-utils/issues/20

### CODE REVIEW

File: stackable-bcrypt/src/main/java/tech/stackable/nifi/BCryptPasswordEncoder.java [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+    private static final int DEFAULT_COST = 12;
+    private static final Version BCRYPT_VERSION;
+    private static final Hasher HASHER;
+    private static final Verifyer VERIFYER;
+
+    public BCryptPasswordEncoder() {
+    }
+
+    public String encode(char[] password) {
+        return HASHER.hashToString(12, password);
```
Comment (teozkr) [inline stackable-bcrypt/src/main/java/tech/stackable/nifi/BCryptPasswordEncoder.java:19]:
```suggestion
        return HASHER.hashToString(DEFAULT_COST, password);
```

File: stackable-bcrypt/README.md [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+
+The tool reads from stdin.
+
+## Usage
+
+stackable-bcrypt [OPTIONS] 
+
+OPTIONS:
+-c, --cost <COST>
+[default: 10]
```
Comment (teozkr) [inline stackable-bcrypt/README.md:15]: Looks like this isn't currently supported by the Java rewrite?

File: stackable-bcrypt/pom.xml [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+        <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
+        <source>11</source>
+        <target>11</target>
+    </properties>
+
+    <dependencies>
+        <dependency>
+            <groupId>at.favre.lib</groupId>
+            <artifactId>bcrypt</artifactId>
+            <version>0.9.0</version>
```
[the remark this answers is not in the document]
Comment (soenkeliebau) [reply stackable-bcrypt/pom.xml:24]: I'm happy to try and override that and see if it still works. Should not be a large issue, unless there were breaking changes in bouncycastle...

File: stackable-bcrypt/README.md (hunk 2 above)
Comment (soenkeliebau) [reply stackable-bcrypt/README.md:15]: true

File: stackable-bcrypt/pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (soenkeliebau) [reply stackable-bcrypt/pom.xml:24]: I've had a very quick look, it seems like bouncycastle is just a test dependency?

File: stackable-bcrypt/README.md (hunk 2 above)
Comment (soenkeliebau) [reply stackable-bcrypt/README.md:15]: fixed

File: stackable-bcrypt/pom.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (teozkr) [reply stackable-bcrypt/pom.xml:24]: Oh :facepalm: 

Nothing to see here then, move right along.

**Review by teozkr — APPROVED** [review]

**Pull request merged**